File: DemoServer/AdditionalLanguages/python/textSearch/fTSWithStaticIndexSingleField/ftsWithStaticIndexSingleField.py
from demo_example import Example, RunParamsBase
from models import Category

#region Usings
from ravendb import AbstractIndexCreationTask
from ravendb.documents.indexes.definitions import FieldIndexing
#endregion

class RunParams(RunParamsBase):
    def __init__(self, search_term: str = None):
        self.search_term = search_term

#region Demo
#region Step_1
class Categories_DescriptionText(AbstractIndexCreationTask):
#endregion
    def __init__(self):
        #region Step_2
        super().__init__()
        self.map = (
            "docs.Products.Select(product => new {"
            "    CategoryDescription = category.Description"
            "})"
        )        
        #endregion

        #region Step_3
        self._index("CategoryDescription", FieldIndexing.SEARCH)
        #endregion
#endregion

class FtsWithStaticIndexSingleField(Example):
    def run(self, run_params: RunParams):
        searchTerm = run_params.search_term
        Categories_DescriptionText().execute(self.document_store_holder.store())

        #region Demo
        with self.document_store_holder.store().open_session() as session:
            #region Step_4
            categories_with_search_term = list(
                session.query_index_type(Categories_DescriptionText, Category).search(
                    "CategoryDescription", searchTerm
                )
            )
            #endregion
        #endregion

        return categories_with_search_term

File: DemoServer/AdditionalLanguages/php/textSearch/fTSWithStaticIndexSingleField/FTSWithStaticIndexSingleField.php
<?php

namespace RavenDB\Demo\textSearch\fTSWithStaticIndexSingleField;

//region Usings
use RavenDB\Documents\Indexes\AbstractIndexCreationTask;
use RavenDB\Documents\Indexes\FieldIndexing;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Category;

//region Demo
//region Step_1
class Categories_DescriptionText extends AbstractIndexCreationTask
{
//endregion
    public function __construct()
    {
        parent::__construct();

        //region Step_2
        $this->map = "docs.Categories.Select(category => new { " .
            "    CategoryDescription = category.Description " .
            "})";
        //endregion

        //region Step_3
        $this->index("CategoryDescription", FieldIndexing::search());
        //endregion
    }
}
//endregion

class FTSWithStaticIndexSingleField
{
    public function __invoke(RunParams $runParams): array
    {
        $searchTerm = $runParams->getSearchTerm();

        //region Demo
        $categoriesWithSearchTerm = [];

        $session = DocumentStoreHolder::getStore()->openSession();
        try {
            //region Step_4
            $categoriesWithSearchTerm = $session->query(Category::class, Categories_DescriptionText::class)
                ->whereEquals("CategoryDescription", $searchTerm)
                ->toList();
            //endregion
        } finally {
            $session->close();
        }
        //endregion

        return $categoriesWithSearchTerm;
    }
}
